.vscode/tasks.json

```diff
@@ -1,32 +1,95 @@
-	"tasks": [
-		{
-			"label": "Compile",
-			"type": "shell",
-			"command": "make -j`nproc` all",
-			"problemMatcher": "$gcc",
-			"group": {
-				"kind": "build",
-				"isDefault": true
-			},
-		},
-		{
-			"label": "Clean",
-			"type": "shell",
-			"command": "make -j`nproc` clean",
-			"problemMatcher": "$gcc",
-			"group": "build",
-		},
-		{
-			"label": "Generate compile_commands.json",
-			"type": "shell",
-			"command": "make clean-build clean-tools && bear -- make -j2 standard",
-			"problemMatcher": "$gcc",
-			"group": "build",
-			"dependsOn": ["Compile"],
-		}
-	]
-}
\ No newline at end of file
+    "version": "2.0.0",
+    "tasks": [
+        {
+            "label": "Check For Game Directory",
+            "type": "shell",
+			"command": "echo -e '\\033[0;97mSource Code directory: ${workspaceFolder}\\033[0m'; if [ ! -d \"${config:game.directory}\" ]; then echo -e '\\033[0;91mGame directory missing in settings.json. Please update it.\\033[0m'; exit 1; else echo -e '\\033[0;97mGame directory: ' ${config:game.directory} '\\033[0m'; fi",
+			"runOptions": {
+                "runOn": "folderOpen"
+            },
+            "group": "build",
+            "presentation": {
+                "echo": false,
+                "focus": true,
+                "panel": "shared",
+                "showReuseMessage": false
+            }
+        },
+        {
+            "label": "Compile",
+            "type": "shell",
+            "command": "make -j`nproc` all && echo -e '\\033[0;32mCompilation successful!\\033[0m'",
+            "dependsOn": "Check For Game Directory",
+            "group": "build",
+            "presentation": {
+                "echo": false,
+                "focus": true,
+                "panel": "shared",
+                "showReuseMessage": false,
+            }
+        },
+        {
+            "label": "Copy Files",
+            "type": "shell",
+			"command": "cd \"${config:game.directory}\" && cp \"${workspaceFolder}/bin/\"* .",
+			"dependsOn": "Compile",
+            "group": "build",
+            "presentation": {
+                "echo": false,
+                "focus": true,
+                "panel": "shared",
+                "showReuseMessage": false,
+            }
+        },
+        {
+            "label": "Run Game",
+            "type": "shell",
+            "command": "cd \"${config:game.directory}\" && if [ -n \"$WSL_DISTRO_NAME\" ]; then cmd.exe /C \"keeperfx.exe ${config:game.arguments}\"; else wine 'keeperfx.exe' ${config:game.arguments}; fi",
+            "dependsOn": "Copy Files",
+            "group": "build",
+            "presentation": {
+                "echo": false,
+                "focus": true,
+                "panel": "shared",
+                "showReuseMessage": false,
+            }
+        },
+        {
+            "label": "Compile, Copy Files, Run Game",
+            "type": "shell",
+            "command": "",
+            "dependsOn": "Run Game",
```

Comment:
This task doesn't do anything more than Run Game, why not merge those 2?

Reply:
I still wouldn't want the game to start every single time I compile but ok.

Reply:
merged them